### PR TITLE
SDIT-937 Add tests using PrisonerBuilder

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/AbstractSearchDataIntegrationTest.kt [hunk 1]
```diff
@@ -29,6 +33,7 @@ import uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto.PossibleM
  * Subclasses will get the same set of search data and we have implemented our own custom junit orderer to ensure that
  * no other type of tests run inbetween and cause a re-index to ruin our data.
  */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/AbstractSearchDataIntegrationTest.kt:36]: This is needed as not all tests use the same test data

File: hmpps-prisoner-search/src/test/resources/application-test.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 server:
   shutdown: immediate
 
+aws:
+  provider: localstack
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/test/resources/application-test.yml:5]: Not sure why this got added in - will remove

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/ObjectBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  }
+
+fun PrisonerBuilder.toPrisoner(): Prisoner =
+  Prisoner(ob = toOffenderBooking(), incentiveLevel = toIncentiveLevel(), null)
+
+private fun getOffenderBookingTemplate(): OffenderBooking =
+  GsonConfig().gson().fromJson("/templates/booking.json".readResourceAsText(), OffenderBooking::class.java)
+
+fun Prisoner(ob: OffenderBooking, incentiveLevel: IncentiveLevelDto?, restrictedPatientData: RestrictivePatient?) =
+  Prisoner().apply { this.translate(null, ob, Result.success(incentiveLevel), restrictedPatientData) }
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/ObjectBuilder.kt:215]: Is the plan to get rid of OffenderBooking at some pont? The search doesn't really need to know about it at all once you've unpicked all the indexing test set up.

**Review by mikehalmamoj — APPROVED** [review]

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/ObjectBuilder.kt (hunk 3 above)
Comment (karenmillermoj) [reply hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/ObjectBuilder.kt:215]: yes - hoping to get that all tidied up once all code is in and working

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/nomis/dto/RestrictivePatient.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.model.nomis.dto
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+data class RestrictivePatient(
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/model/nomis/dto/RestrictivePatient.kt:6]: Will update - cheers

**Pull request merged**